### ISSUE TITLE
Game Request: Kimaten (Redirect font)

### ISSUE THREAD

**Issue by Gygarde** [marcussacana/Specific-Games#28]
Hello!

I want to redirect the font of this game. It uses Silky Engine (Same as Nanairo). .exe is [here](https://cdn.discordapp.com/attachments/754950553590825032/911670834890944572/kimaten.zip).

Let me know if you need something.

Thank you.

**Comment by marcussacana** [comment]
Check if the SRL is loaded using the d3d9.dll as wrapper https://github.com/marcussacana/StringReloads

**Comment by Gygarde** [comment]
@marcussacana It seems it detected it, but the game won't launch.

LOG:
[INF] SRL - StringReloads v7.5.2
[INF] Created by Marcussacana
[INF] SRL Initialized

**Comment by marcussacana** [comment]
Should be because of the missing database, delete the Cache.srl file and create an empty Strings.lst file in the same folder of the Cache.srl file.

**Comment by Gygarde** [comment]
Hmm, it just crashed

**Comment by Gygarde** [comment]
@marcussacana I tried with several configurations in SRL, but no luck. Please check my file:

> [StringReloads]
> AutoInstall=true
> BreakLine=\n
> CacheOutput=true
> Debug=false
> Dump=false
> DumpRegex=false
> Encoding=SJIS
> FastMode=false
> Hashset=false
> HeapAlloc=false
> ImportHook=false
> ImportHookEx=false
> LoadLocalFonts=false
> Log=true
> LogFile=true
> LogLevel=INF
> Overwrite=false
> ReloadRegexCaptures=true
> SanityCheck=true
> TrimMatch=false
> WrapperPatch=true
> Workspace=StringReloads
> 
> [Wordwrap]
> UseRelativeWidth=false
> RelativeWidth=if([Width]-([Width]/5) < 30, 50, [Width]-([Width]/5))
> DefaultWidth=50
> 
> [Hooks]
> CreateFontA=false
> CreateFontW=false
> CreateFontIndirectA=false
> CreateFontIndirectW=false
> ExtTextOutA=false
> ExtTextOutW=false
> GetCharABCWidthsFloatA=false
> GetCharABCWidthsFloatW=false
> GetCharacterPlacementA=false
> GetCharacterPlacementW=false
> GetGlyphOutlineA=false
> GetGlyphOutlineW=false
> GetTextExtentPoint32A=false
> GetTextExtentPoint32W=false
> lstrcpyA=false
> lstrcpyW=false
> MultiByteToWideChar=false
> SysAllocString=false
> WideCharToMultiByte=false
> TextOutA=false
> TextOutW=false
> 
> [MultiByteToWideChar]
> AutoEncoding=true
> CodePage=-1
> RemapAlt=false
> UndoRemap=false
> 
> [WideCharToMultiByte]
> AutoEncoding=true
> CodePage=-1
> RemapAlt=false
> UndoRemap=false
> 
> [TextOutA]
> RemapAlt=false
> UndoRemap=false
> 
> [TextOutW]
> RemapAlt=false
> UndoRemap=false
> 
> [ExtTextOutA]
> RemapAlt=false
> UndoRemap=false
> 
> [ExtTextOutW]
> RemapAlt=false
> UndoRemap=false
> 
> [GetCharABCWidthsFloatA]
> RemapAlt=false
> UndoRemap=false
> 
> [GetCharABCWidthsFloatW]
> RemapAlt=false
> UndoRemap=false
> 
> [GetCharacterPlacementA]
> RemapAlt=false
> UndoRemap=false
> 
> [GetCharacterPlacementW]
> RemapAlt=false
> UndoRemap=false
> 
> [GetGlyphOutlineA]
> RemapAlt=false
> UndoRemap=false
> 
> [GetGlyphOutlineW]
> RemapAlt=false
> UndoRemap=false
> 
> [GetTextExtentPoint32A]
> RemapAlt=false
> UndoRemap=false
> 
> [GetTextExtentPoint32W]
> RemapAlt=false
> UndoRemap=false
> 
> [Modifiers]
> Escape=false
> MonoWordWrap=false
> Remaper=false
> RemaperAlt=false
> 
> [Mods]
> CoInitializeFix=false
> ForceExit=false
> PatchRedir=false
> 
> [Trim]
> Prefixes=\\wc\n\\ws
> Sufixes=\\wc\n\\ws
> 
> [Filter]
> AcceptableRange=0-9A-Za-zÀ-ÃÇ-ÎÓ-ÕÚ-Ûà-ãç-îó-õú-û｡-ﾟ!?~.,''""
> AllowNumbers=false
> DenyList=@\n§\n_\n<\n>\n/\n[\n]\n=
> DumpFilter=true
> DumpRegexFilter=false
> DumpAcceptableRange=false
> FromAsian=false
> IgnoreList=\n､\n｡
> QuoteList=<>\n()\n[]\n“”\n［］\n《》\n«»\n「」\n『』\n【】\n（）\n''\n""
> RegexFilter=false
> Sensitivity=4
> UseDB=true
> 
> [Font.0]
> From=*
> ;FromWidth=0
> ;FromHeight=0
> ;FromCharset=0
> To=Gothic Modded
> Charset=0
> Width=+0
> Height=+0

**Comment by marcussacana** [comment]
but you tried delete the srl file and create the lst one?

**Comment by marcussacana** [comment]
Also, test rename the d3d9.dll to dxgi.dll or dinput8.dll

**Comment by Gygarde** [comment]
@marcussacana Yes, I tried. 
![imagen](https://user-images.githubusercontent.com/38927751/142750465-6d61ab6b-f465-4bf6-b973-cd6d0d97a1ab.png)
Also tried with all wrappers, works  only with d3d9, but it crashes without starting. The game is x32, so I'm using the correct version of SRL (x86).

**Comment by marcussacana** [comment]
Yep, then should require manual patching in the game executable... hmm
I will take a look when I can, if the game have any .dll file in his directory upload to me too.

**Comment by Gygarde** [comment]
Just some from emote, but here you go anyway.
[kimaten.zip](https://github.com/marcussacana/Specific-Games/files/7575790/kimaten.zip)

Thank you.

**Comment by Gygarde** [comment]
@marcussacana Just an update: I can't use symbols inside the original script (the game just crash), so I think it needs more than just a font redirection. Do you think would be possible to use a DB (.lst) as usual?

**Comment by marcussacana** [comment]
You can't put japanese characters?

**Comment by Gygarde** [comment]
@marcussacana nope, the game has mapped characters. And I would need more useless characters to replace them with mine. (Adding more result in crash, btw)

` !"%'(),-.123456789
:?ABCDEFGHIJKLMNOPQR
STUVWXYZ\abcdefghijk
lmnopqrstuvwxyz·×—`

**Comment by marcussacana** [comment]
What tool you're using?

**Comment by Gygarde** [comment]
I extract with Garbro.
This for .MAP (Scripts): https://github.com/TesterTesterov/MAPTool/blob/main/MAPTool.py

**Comment by marcussacana** [comment]
I never found a game using this file before, then I really don't know what we can do here.
The game don't accept japanese characters, this mean you're trying translate a english version of the game?
You tried make your patch based in the japanese release?

**Comment by Gygarde** [comment]
Yeah, I'm trying with the english release. Maybe I can try to modify the japanese mapped characters, but I think I'm running out of options.

Edit: Tried, no luck.

**Comment by Cosetto** [comment]
@Gygarde The new Silky's engine uses bitmap font, you need to find the font file somewhere like this 
![image](https://user-images.githubusercontent.com/81801486/184095815-012d72ac-8ba4-4ed9-84a9-26adc48f65ff.png)
And for .MAP use [this](https://github.com/arcusmaximus/VNTranslationTools)
